rclcpp/test/rclcpp/test_service_introspection.cpp

```diff
@@ -80,6 +81,7 @@ class TestServiceIntrospection : public ::testing::Test
   rclcpp::Node::SharedPtr node;
   rclcpp::Client<BasicTypes>::SharedPtr client;
   rclcpp::Service<BasicTypes>::SharedPtr service;
+  rclcpp::QoS qos_profile_ = rclcpp::QoS(10).reliable();
```

Comment:
Having the qos reliable and depth 10 for both publishers and subscription for service events to make this test result stable.

Unfortunately even with this setting, it still fails with `rmw_connextdds`.
I tried to increase the timeout from `1000` to `3000` but still sometimes fails.

```bash
colcon test --event-handlers console_direct+ --packages-select rclcpp --ctest-args -R test_service_introspection --retest-until-fail 100
...
74: /root/ros2_ws/colcon_ws/src/ros2/rclcpp/rclcpp/test/rclcpp/test_service_introspection.cpp:200: Failure
74: Expected equality of these values:
74:   events.size()
74:     Which is: 1
74:   2U
74:     Which is: 2
74:
74: [  FAILED  ] TestServiceIntrospection.service_introspection_enable_disable_events (2312 ms)
74: [ RUN      ] TestServiceIntrospection.service_introspection_enable_disable_event_content
74: RTI Connext DDS Non-commercial license is for academic, research, evaluation and personal use only. USE FOR COMMERCIAL PURPOSES IS PROHIBITED. See RTI_LICENSE.TXT for terms. Download free tools at rti.com/ncl. License issued to Non-Commercial User license@rti.com For non-production use only.
74: Expires on 00-jan-00 See www.rti.com for more information.
74: [       OK ] TestServiceIntrospection.service_introspection_enable_disable_event_content (216 ms)
74: [----------] 3 tests from TestServiceIntrospection (2834 ms total)
74:
74: [----------] Global test environment tear-down
74: [==========] 3 tests from 1 test suite ran. (2843 ms total)
74: [  PASSED  ] 2 tests.
74: [  FAILED  ] 1 test, listed below:
74: [  FAILED  ] TestServiceIntrospection.service_introspection_enable_disable_events
74:
74:  1 FAILED TEST
74: -- run_test.py: return code 1
74: -- run_test.py: inject classname prefix into gtest result file '/root/ros2_ws/colcon_ws/build/rclcpp/test_results/rclcpp/test_service_introspection.gtest.xml'
74: -- run_test.py: verify result file '/root/ros2_ws/colcon_ws/build/rclcpp/test_results/rclcpp/test_service_introspection.gtest.xml'
    Test #74: test_service_introspection .......***Failed    2.91 sec

0% tests passed, 1 tests failed out of 1

Label Time Summary:
gmock     =   2.91 sec*proc (1 test)
mimick    =   2.91 sec*proc (1 test)

Total Test time (real) =   8.89 sec

The following tests FAILED:
         74 - test_service_introspection (Failed)
Errors while running CTest
Output from these tests are in: /root/ros2_ws/colcon_ws/build/rclcpp/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
--- stderr: rclcpp
Errors while running CTest
Output from these tests are in: /root/ros2_ws/colcon_ws/build/rclcpp/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
---
Finished <<< rclcpp [9.07s]     [ with test failures ]

Summary: 1 package finished [10.1s]
  1 package had stderr output: rclcpp
  1 package had test failures: rclcpp
```